source/Calamari.Tests/Fixtures/Conventions/ContributeVariablesFromJsonFileConventionFixture.cs

```diff
@@ -0,0 +1,114 @@
+
+            foreach (var variable in newVariables)
+            {
+                fileVariables.Add(variable.Key, variable.Value);
+            }
+            
+            var convention = new ContributeVariablesFromJsonFileConvention(path => fileVariables);
+            
+            convention.Install(deployment);
+
+            var actual = deployment.Variables;
+
+            assertions(actual);
+        }
+
+        [Test]
+        public void VariablesAreNotContributedIfEnvVarIsNotFound()
+        {
+            RunTest(
+                new (string, string)[0],
+                new[]
+                {
+                    ("new.key", "new.value")
+                },
+                actual => actual.Should().BeEmpty()
+            );
+        }
+
+        [Test]
+        public void VariablesAreContributedIfEnvVarIsFound()
+        {
+            RunTest(
+                new[]
+                {
+                    (ContributeVariablesFromJsonFileConvention.AdditionalVariablesKey, "ignored_by_this_test"),
+                    ("existing.key", "existing.value")
+                },
+                new[]
+                {
+                    ("new.key", "new.value")
+                },
+                actual =>
+                {
+                    actual.Should().HaveCount(3);
+                    actual.Should().ContainSingle(pair => 
+                        pair.Key == ContributeVariablesFromJsonFileConvention.AdditionalVariablesKey &&
+                        pair.Value == "ignored_by_this_test");
+                    actual.Should().ContainSingle(pair => 
+                        pair.Key == "existing.key" &&
+                        pair.Value == "existing.value");
+                    actual.Should().ContainSingle(pair => 
+                        pair.Key == "new.key" &&
+                        pair.Value == "new.value");
+                }
+            );
+        }
+
+        [Test]
+        public void ThrowsExceptionIfFileNotFound()
+        {
+            const string filePath = "c:/assuming/that/this/file/doesnt/exist.json";
+            
+            var deployment = new RunningDeployment(
+                null,
+                new CalamariVariableDictionary
+                {
+                    { ContributeVariablesFromJsonFileConvention.AdditionalVariablesKey, filePath }
+                }
+            );
+
+            var convention = new ContributeVariablesFromJsonFileConvention();
+
+            convention
+                .Invoking(c => c.Install(deployment))
+                .Should()
+                .Throw<CommandException>()
+                // Make sure that the message says how to turn this feature off.
+                .Where(e => e.Message.Contains(ContributeVariablesFromJsonFileConvention.AdditionalVariablesKey))
+                // Make sure that the message says where it looked for the file.
+                .Where(e => e.Message.Contains(filePath));
```

Comment:
Consider making this an exact match on the expected exception

Reply:
Sorry, I forgot to circle back to this when we discussed the PR. My goal was to strike a balance between ensuring that all the useful stuff is in the exception message while also avoiding creating a brittle test.

Reply:
I can see your point.
As its user facing though, I'd be okay with a little brittleness.
Your call.